File: leet/0083-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
    last := head
    
    for last != nil {
        next := last.Next
        
        if next != nil && next.Val == last.Val{
            last.Next = next.Next
            continue
        }
        
        last = next
    }
    
    return head
}

File: leet/0083-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    let last = head
    
    while (last){
        let cur = last
        while(cur && cur.val == last.val)
            cur = cur.next
        
        last.next = cur
        last = last.next
    }
    
    return head
};
